include/NovelRT/Exceptions/NullPointerException.h

```diff
@@ -0,0 +1,21 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_NULL_POINTER_EXCEPTION_EXCEPTION_H
+#define NOVELRT_NULL_POINTER_EXCEPTION_EXCEPTION_H
+
+#include <stdexcept>
+#include <string>
+
+namespace NovelRT::Exceptions {
+  class NullPointerException final : public std::runtime_error {
+  public:
+  NullPointerException() :
+      std::runtime_error("Pointer does not point to an instance.")
```

Comment:
I feel like the message would make more sense if it said "Pointer does not point to a valid object instance." Or maybe something like C#'s NRE message like "Object pointer not set to an instance of an object."?